### PR TITLE
nixos/dovecot2: turn `mailboxes`-option into an attr-set

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

* 7922a78: add the `autoexpunge`-setting to `services.dovecot2.mailboxes`. With this option specified, email older than the specified time will be automatically removed.

  This is e.g. helpful for spam-folders.

* 907d046 : turned the `mailboxes`-option into an attr-set. This makes it easier to override the settings for a single mailbox.

  It's still possible to use a list for backwards-compatibility, but this will raise a deprecation warning.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by cole-h — APPROVED** [review]
Small nit, else LGTM. Though, I don't use dovecot, so I can't verify this works... I'm trusting you already did that ;)

**Comment by Ma27** [comment]
Rebased onto latest master.

File: nixos/doc/manual/release-notes/rl-2009.xml [hunk 1]
```diff
@@ -614,6 +614,29 @@ systemd.services.nginx.serviceConfig.ReadWritePaths = [ "/var/www" ];
      queued on the kernel side of the netlink socket.
     </para>
    </listitem>
+   <listitem>
+    <para>
+     Specifying <link linkend="opt-services.dovecot2.mailboxes">mailboxes</link> in the <package>dovecot2</package>-module
```
Comment (nlewo) [inline nixos/doc/manual/release-notes/rl-2009.xml:619]:
```suggestion
     Specifying <link linkend="opt-services.dovecot2.mailboxes">mailboxes</link> in the <package>dovecot2</package> module
```

**Review by nlewo — COMMENTED** [review]
:+1:  for this change.

I think we will remove the list format for the release 21.03. What about announcing this date in the warning, release notes and adding a comment close to this code?

**Comment by Ma27** [comment]
@nlewo fixed :)

**Comment by nlewo** [comment]
@GrahamcOfBorg test dovecot

**Review by nlewo — APPROVED** [review]

**Comment by nlewo** [comment]
Thanks!

**Pull request merged**